### PR TITLE
Build akka-cluster-sharding with scala3

### PR DESCRIPTION
**Pull request by raboof**
Umbrella issue https://github.com/akka/akka/issues/30243
Not running tests yet due to https://github.com/akka/akka/issues/30711

### CODE REVIEW

File: akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingSpec.scala [hunk 1]
```diff
@@ -117,6 +117,8 @@ abstract class ClusterShardingSpecConfig(
     val entityRecoveryStrategy: String = "all")
     extends MultiNodeClusterShardingConfig(mode = mode, rememberEntitiesStore = rememberEntitiesStore) {
 
+  val theMode = mode
```
Comment (johanandren) [inline akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingSpec.scala:120]: ?

File: akka-cluster-sharding/src/test/scala/akka/cluster/sharding/GetShardTypeNamesSpec.scala [hunk 2]
```diff
@@ -37,7 +37,7 @@ class GetShardTypeNamesSpec extends AkkaSpec(GetShardTypeNamesSpec.config) with
 
   "GetShardTypeNames" must {
     "contain empty when join cluster without shards" in {
-      ClusterSharding(system).shardTypeNames should ===(Set())
+      ClusterSharding(system).shardTypeNames should ===(Set.empty[String])
```
Comment (johanandren) [inline akka-cluster-sharding/src/test/scala/akka/cluster/sharding/GetShardTypeNamesSpec.scala:40]: Hehe, nice with the previous "compare this but only if it is the exact same type" and then "oh, by the way, infer this right hand type for me before you compare" :D

**Review by johanandren — APPROVED** [review]
LGTM with one question mark (left over statement?)

**Comment by raboof** [comment]
(failure was #30664)

**Pull request merged**